src/Log/Store.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * This file is part of the Global Trading Technologies Ltd doctrine-auditable-bundle package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types = 1);
+
+namespace Gtt\Bundle\DoctrineAuditableBundle\Log;
+
+use Doctrine\Bundle\DoctrineBundle as Doctrine;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Gtt\Bundle\DoctrineAuditableBundle\Exception;
+use ProxyManager\Proxy\ValueHolderInterface;
+use function array_key_exists, get_class, spl_object_hash;
+
+/**
+ * Collects entities changes' description to be saved as changelog description for managed entities by auditable
+ *
+ * This is more cleaner alternative for retrieving changelog comment from entity (
+ * because it's application layer violation)
+ */
+class Store
+{
+    /**
+     * Stack of entities' changes description
+     *
+     * @var array
+     *
+     * @example [
+     *             'entityManager_spl_object_hash' => [
+     *               'entity_spl_object_hash' => 'changelog comment'
+     *             ]
+     *          ]
+     */
+    private $store;
+
+    /**
+     * @var Doctrine\Registry
+     */
+    private $doctrineRegistry;
+
+    /**
+     * Logger constructor.
+     *
+     * @param Doctrine\Registry $doctrineRegistry
+     */
+    public function __construct(Doctrine\Registry $doctrineRegistry)
+    {
+        $this->doctrineRegistry = $doctrineRegistry;
+        $this->store            = [];
+    }
+
+    /**
+     * Describes entity changes
+     *
+     * @param object                 $entity        Entity
+     * @param string                 $comment       Comment that describes given entity changes
+     * @param EntityManagerInterface $entityManager Entity manager that manages given entity
+     *
+     * @throws Exception\LogicException When no related entity manager found for given entity
+     */
+    public function describe(object $entity, string $comment): void
+    {
+        if (empty($comment)) {
```

Comment:
I don't like this `if` block, maybe just remove it and force user to give proper description or not to call this method at all.

Reply:
it's weird to describe entity changes with empty value (at this point old strategy may do so, and i guess should be changed to as same as it implemented here in next major update) so i think this check should be saved, this condition just saves some processing time & do not allow to execute excess operations